src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -423,17 +441,22 @@ private void ExecuteCompilationActionsCore(ImmutableArray<CompilationAnalyzerAct
             }
 
             var addDiagnostic = GetAddDiagnostic(semanticModel.SyntaxTree, analyzer, isSyntaxDiagnostic: false);
+            Func<Diagnostic, bool> isSupportedDiagnostic = d => IsSupportedDiagnostic(analyzer, d);
```

Comment:
@heejaechang Do you find this an acceptable approach for this specific consideration as-is? If not, do you have a particular preference regarding refactoring now or just filing an issue for later? So far I didn't pursue this further after I addressed the main items that PerfView was showing me, but this code is in your main area so I'll take care of it as you see fit.